polyfill/lib/duration.mjs

```diff
@@ -453,6 +453,9 @@ export class Duration {
       relativeTo
     ));
     ({ years, months, weeks, days } = ES.BalanceDurationRelative(years, months, weeks, days, largestUnit, relativeTo));
+    if (ES.IsTemporalZonedDateTime(relativeTo)) {
+      relativeTo = ES.MoveRelativeZonedDateTime(relativeTo, years, months, weeks, 0);
```

Comment:
I admit I don't fully understand the logic in this PR so I'm having trouble reviewing it. I'm also concerned about the failing tests. I'm happy to jump on a call over the weekend (or even later tonight depending on the time) if you want to go over this 1:1 like we did a few days ago if that'd help.

Reply:
I feel like I'm close to solving the failing tests, by taking the approach of going step by step through what would happen if calling `relativeTo.until(relativeTo.add(duration), options)`. We certainly don't have to merge this until I've fixed the two remaining tests, but I did want to get the interim progress cleaned up and ready for review.

I'm trying to describe the rationale for each change in the commit messages but at this point I'm probably not able to take enough of a step back to write very clearly about it, so maybe walking through it is a good idea. I'm hoping to fix this last test quickly and spend the rest of the weekend relatxing though, so we'll have to see how it goes.

Reply:
OK, I'm around for most of the weekend and would be happy to walk through with you, just let me know.